akka-http-bench-jmh/src/main/scala/akka/http/impl/engine/http2/H2ServerProcessingBenchmark.scala

```diff
@@ -5,55 +5,53 @@
 package akka.http.impl.engine.http2
 
 import java.util.concurrent.{ CountDownLatch, TimeUnit }
-
+import scala.collection.immutable
 import scala.concurrent.Await
-import scala.concurrent.Future
 import scala.concurrent.duration._
 import akka.actor.ActorSystem
 import akka.http.CommonBenchmark
-import akka.http.impl.engine.http2.FrameEvent.HeadersFrame
-import akka.http.impl.engine.http2.framing.FrameRenderer
-import akka.http.scaladsl.model.HttpResponse
-import akka.http.scaladsl.settings.ServerSettings
+import akka.http.scaladsl.client.RequestBuilding.Get
+import akka.http.scaladsl.model.HttpEntity.LastChunk
+import akka.http.scaladsl.model.headers.RawHeader
+import akka.http.scaladsl.model.{ ContentTypes, HttpEntity, HttpRequest, HttpResponse }
 import akka.stream.ActorMaterializer
-import akka.stream.TLSProtocol.{ SslTlsInbound, SslTlsOutbound }
 import akka.stream.scaladsl.{ Flow, Keep, Sink, Source }
-import akka.util.ByteString
 import com.typesafe.config.ConfigFactory
 import org.openjdk.jmh.annotations._
 
+import scala.util.{ Failure, Success }
+
 class H2ServerProcessingBenchmark extends CommonBenchmark {
-  // Obtained by converting the input request bytes from curl with --http2-prior-knowledge
-  def request(streamId: Int) =
-    FrameRenderer.render(HeadersFrame(streamId, endStream = true, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman, None))
 
-  val response: HttpResponse = HPackSpecExamples.FirstResponse
+  @Param(Array("closedelimited", "chunked"))
+  var responsetype: String = _
+
+  var response: HttpResponse = _
 
-  var httpFlow: Flow[ByteString, ByteString, Any] = _
+  var httpFlow: Flow[HttpRequest, HttpResponse, Any] = _
   implicit var system: ActorSystem = _
   implicit var mat: ActorMaterializer = _
 
-  val packedResponse = ByteString(-62, -63, -64, -65, -66)
-
   val numRequests = 10000
 
   @Benchmark
   @OperationsPerInvocation(10000) // should be same as numRequest
   def benchRequestProcessing(): Unit = {
+    implicit val ec = system.dispatcher
+
     val latch = new CountDownLatch(numRequests)
 
     val requests =
-      Source(Http2Protocol.ClientConnectionPreface +: Range(0, numRequests).map(i => request(1 + 2 * i)))
+      Source.repeat(Get("/")).take(numRequests)
```

Comment:
Does this benchmark really work? It seems to spend all it's time in the materialization inside the bench method.